components/application-gateway/internal/proxy/unauthorizedresponseretrier_test.go

```diff
@@ -55,7 +55,10 @@ func TestForbiddenResponseRetrier_CheckResponse(t *testing.T) {
 		req, err := http.NewRequest(http.MethodGet, "/orders/123", nil)
 		require.NoError(t, err)
 
-		rr := newUnauthorizedResponseRetrier("id1", req, 10, updateCacheEntryFunction)
+		secondRequestBody, err := createSecondRequestBody(req)
```

Comment:
The request body is `nil` here, we do not need to copy it. The same is for other cases.

Reply:
Done!